frontend/components/site/SitePublishedFilesTable.jsx

```diff
@@ -1,41 +1,141 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import autoBind from 'react-autobind';
 
 import publishedFileActions from '../../actions/publishedFileActions';
 import LoadingIndicator from '../LoadingIndicator';
 import AlertBanner from '../alertBanner';
 
 class SitePublishedFilesTable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentPage: 0,
+      filesByPage: {},
+      lastPage: null,
+    };
+  }
+
   componentDidMount() {
     const site = { id: this.props.params.id };
     const branch = this.props.params.name;
-    publishedFileActions.fetchPublishedFiles(site, branch);
+
+    const startAtKey = null; // start without a startAtKey
+    publishedFileActions.fetchPublishedFiles(site, branch, startAtKey);
+
+    autoBind(this, 'previousPage', 'nextPage', 'shouldDisablePreviousPage',
+             'shouldDisableNextPage', 'shouldShowButtons');
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const publishedFiles = nextProps.publishedFiles;
+
+    if (publishedFiles.data && !publishedFiles.isLoading) {
+      const files = publishedFiles.data.files || [];
+
+      // save the current page of files state
+      const filesByPage = this.state.filesByPage;
+      const currentPage = this.state.currentPage;
+
+      filesByPage[currentPage] = files;
+      this.setState({ filesByPage });
+
+      if (!publishedFiles.data.isTruncated) {
+        this.setState({ lastPage: currentPage });
+      }
+    }
+  }
+
+  shouldDisablePreviousPage() {
+    return this.state.currentPage === 0;
```

Comment:
`return !this.state.currentPage`, unless we need to know if `currentPage` has been defined.